system/lib/sbrk.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ *
+*/
+
+#ifndef EMSCRIPTEN_NO_ERRNO
+#include <errno.h>
+#endif
+#include <stddef.h>
+#include <stdint.h>
+
+#define WASM_PAGE_SIZE 65536
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern intptr_t* emscripten_get_sbrk_ptr(void);
+extern int emscripten_resize_heap(size_t requested_size);
+extern size_t emscripten_get_heap_size(void);
+
+#ifdef __cplusplus
+}
+#endif
+
+void *sbrk(intptr_t increment) {
+  intptr_t* sbrk_ptr = emscripten_get_sbrk_ptr();
+  intptr_t old_brk = *sbrk_ptr;
+  // TODO: overflow checks
+  intptr_t new_brk = old_brk + increment;
+#if __wasm__
```

Comment:
Shouldn't this be `#ifdef __wasm__`? I don't think fastcomp defines `__wasm__` == 0?